include/c74_min_timer.h

```diff
@@ -17,39 +17,44 @@ namespace c74::min {
     template<timer_options options = timer_options::deliver_on_scheduler>
     class timer;
 
-    extern "C" void timer_tick_callback(timer<>* an_owner);    // defined in c74_min_impl.h
-    extern "C" void timer_qfn_callback(timer<>* a_timer);      // defined in c74_min_impl.h
+	class timer_base;
 
+    struct timer_impl {
+        max::t_object m_obj;
+        timer_base* m_owner;
+    };
 
-    /// The timer class allows you to schedule a function to be called in the future using Max's scheduler.
-    /// Note: the name `timer` was chosen instead of `clock` because of the use of the type is `clock` is ambiguous on
-    /// the Mac OS when not explicitly specifying the `c74::min` namespace.
-    /// @tparam		options		Optional argument to alter the delivery from the scheduler thread to the main thread.
-    ///
-    ///	@seealso	#time_value
-    /// @seealso	#queue
-    /// @seealso	#fifo
+    static max::t_class* s_timer_impl_class = nullptr;
 
-    template<timer_options options>
-    class timer {
-    public:
-        /// Create a timer.
-        /// @param	an_owner	The owning object for the timer. Typically you will pass `this`.
-        /// @param	a_function	A function to be executed when the timer is called.
-        ///						Typically the function is defined using a C++ lambda with the #MIN_FUNCTION signature.
+    extern "C" void timer_tick_callback(timer_impl* an_owner);    // defined in c74_min_impl.h
+    extern "C" void timer_qfn_callback(timer_impl* a_timer);      // defined in c74_min_impl.h
 
-        timer(object_base* an_owner, const function a_function)
+	class timer_base {
+	public:
+        timer_base(object_base* an_owner, timer_options options, const function a_function)
         : m_owner { an_owner }
         , m_function { a_function } {
-            m_instance = max::clock_new(this, reinterpret_cast<max::method>(timer_tick_callback));
+
+            if (!s_timer_impl_class) {
+                s_timer_impl_class = max::class_new("min_timer_impl", (max::method)0, (max::method)0, sizeof(timer_impl), 0, NULL);
+                max::class_register(max::CLASS_NOBOX, s_timer_impl_class);
+            }
```

Comment:
if we have 2 different externals that use min, is this a problem? I need to read up about the static but I'd assume each would have their own copy and so each would register this class..
How about if they're using 2 different versions of min that both provide this capability?

Maybe min should lookup the class before trying to register it? Maybe it should encode a version into the class name?

Reply:
Ooooh yes, very good point! Thanks!

The `min_timer_impl` class doesn't really do much. So I don't think there's a clash when mixing different versions of min.

I'm not sure how `class_new` behaves when calling it multiple times for the same class. @jkcmusork - what do you think? Is a class lookup the best way to handle this?

Reply:
maybe its worth adding a comment to indicate that if there is a change in the struct or methods, you'll need to change the name.